pkg/api/validation/validation.go

```diff
@@ -266,6 +266,28 @@ func validateLifecycle(lifecycle *api.Lifecycle) errs.ValidationErrorList {
 	return allErrs
 }
 
+// TODO(dchen1107): Move this along with other defaulting values
+func validatePullPolicyWithDefault(ctr *api.Container) errs.ValidationErrorList {
+	allErrors := errs.ValidationErrorList{}
+
+	if len(ctr.ImagePullPolicy) == 0 {
+		parts := strings.SplitN(ctr.Image, ":", 2)
+		// Check image tag
+		if len(parts) == 2 && parts[1] == "latest" {
```

Comment:
what about images that specify a registry?  e.g. foo.bar.com:5000/my/image:latest